core/src/main/java/org/opensearch/sql/planner/physical/HighlightOperator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.physical;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.opensearch.sql.common.utils.StringUtils;
+import org.opensearch.sql.data.model.ExprTupleValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.model.ExprValueUtils;
+import org.opensearch.sql.expression.DSL;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.ReferenceExpression;
+import org.opensearch.sql.expression.env.Environment;
+
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+import static org.opensearch.sql.data.type.ExprCoreType.STRUCT;
+import static org.opensearch.sql.expression.env.Environment.extendEnv;
+
+/**
+ * Common method actions for ml-commons related operators.
+ */
+
+@Getter
+@EqualsAndHashCode
+public class HighlightOperator extends PhysicalPlan {
+  @Getter
+  private final PhysicalPlan input;
+  @Getter
+  private final Expression highlight;
+
+  public HighlightOperator(PhysicalPlan input, Expression highlight) {
+    this.input = input;
+    this.highlight = highlight;
+  }
+
+  @Override
+  public <R, C> R accept(PhysicalPlanNodeVisitor<R, C> visitor, C context) {
+    return visitor.visitHighlight(this, context);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return input.hasNext();
+  }
+
+  @Override
+  public ExprValue next() {
+    ExprValue inputValue = input.next();
+    Map<String, ExprValue> evalMap = mapHighlight(inputValue.bindingTuples());
+
+    if (STRUCT == inputValue.type()) {
+      ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+      Map<String, ExprValue> tupleValue = ExprValueUtils.getTupleValue(inputValue);
+      for (Map.Entry<String, ExprValue> valueEntry : tupleValue.entrySet()) {
+        if (evalMap.containsKey(valueEntry.getKey())) {
```

Comment:
Could you add a variable for `valueEntry.getKey()`? It will make the code easier to follow.